controller/tests/array_action/svc/array_mediator_svc_test.py

```diff
@@ -228,10 +270,14 @@ def _prepare_lsvdisk_to_return_mapless_target_volume(self):
     def _prepare_lsvdisk_to_return_none(self):
         self.svc.client.svcinfo.lsvdisk.return_value = self._mock_cli_object(None)
 
-    def _prepare_mocks_for_get_snapshot(self):
+    def _prepare_returns_snapshot(self):
         target_cli_vol = self._get_mapped_target_cli_vol()
         self.svc.client.svcinfo.lsvdisk.return_value = self._mock_cli_object(target_cli_vol)
 
+    def _prepare_mocks_for_get_snapshot(self):
+        self._prepare_returns_snapshot()
+        self.fcmaps[0].copy_rate = "0"
```

Comment:
what does `_prepare_returns_snapshot` mean? why separate?

Reply:
because the method "get_snapshot" need spacial preparations..

Reply:
thanks for answering the second question.
yes, I already assume there is some "special" difference that required this separation.
could you please elaborate about this preparations difference?
and again, could you please explain what is the purpose of `_prepare_returns_snapshot`? I couldn't understand it from the name.
for example, why shouldn't `_prepare_returns_snapshot` assign `copy_rate = "0"` inside?

Reply:
the methods` _prepare_mocks_for_get_snapshot` are the one who calls to get_snapshot which validating it is a snapshot by checking the` copy_rate` of its fcmap as target.
the methods who calls `_prepare_returns_snapshot`  are the delete_snapshot do not have that check and will fail when they try to delete due to fcmap as source with `copy_rate = "0"`

Reply:
I changed the name to `_prepare_mocks_for_delete_snapshot`

Reply:
got it. thanks!